### PR TITLE
fix: disable cache maps when "Use Object Cache" is disabled

### LINKED ISSUE #229: Cache Maps should not be stored if "Use Object Cache" setting is not enabled
WPGraphQL Smart Cache currently collects and stores cache maps, intended for local cache (object/transient) invalidation. These maps are being stored _always_, not just when "Use Object Cache" is enabled. 

Folks using Network Cache (WP Engine's EverCache, Varnish, Litespeed, etc) have no use for these cache maps, but they're being stored regardless. 

This is problematic and appears to be causing "out of memory" issues.

Here's a summary of the problem:  

- the `WPGraphQL\SmartCache\Cache\Collection` class is initiated when WordPress loads: https://github.com/wp-graphql/wp-graphql-smart-cache/blob/main/wp-graphql-smart-cache.php#L201-L203
- This hooks into `graphql_return_response` for _every_ GraphQL Query to persist the response: https://github.com/wp-graphql/wp-graphql-smart-cache/blob/main/src/Cache/Collection.php#L18
- Every time a GraphQL Request is executed, the request is added to the map: https://github.com/wp-graphql/wp-graphql-smart-cache/blob/main/src/Cache/Collection.php#L64-L95

This means that anytime a unique GraphQL Query (+variables) is executed, this map grows. 

Since the [$request_key](https://github.com/wp-graphql/wp-graphql-smart-cache/blob/main/src/Cache/Collection.php#L65C25-L65C34) is based on the `queryId`, `query document`, `variables` and `current user`, this leads to a wide array of permutations of query keys. 

i.e. `{ "query": 'query getPost( $id: ID! ) { post( id: $id ) { title } }', "variables": { id: 123 } }`, and `{ "query": 'query getPost( $id: ID! ) { post( id: $id ) { title } }', "variables": { id: 456 } }` generate different keys and are added to the map. 

For higher-traffic / higher-content sites, this can lead to a build-up of the map quite quickly and ultimately leads to memory exhaustion. 

----

### Disable storing the maps if "Use Object Cache" is not enabled

Since the map that is stored is only used for purging cached responses from object cache/transient cache (and not network cache) we should disable the map storage for users that do not have "Use Object Cache" enabled (similar to: https://github.com/wp-graphql/wp-graphql-smart-cache/blob/main/src/Cache/Results.php#L114-L116). (Note, the mapping is _also_ used for our tests, at least as of now, so tests will need to enable this feature)

By default, if "Use Object Cache" is disabled, storing the maps should also be disabled. However there should be a filter to allow the maps to be stored, even if "Use Object Cache" is disabled. This will allow our tests to work _and_ will allow other plugins to enable the maps. For example, if someone was using GET requests on a host that cannot tag the caches with the X-GraphQL-Keys headers, they could take advantage of the maps and purge that way (like we did in the early days of this plugin)

### Reduce the expiration of the local cache  (address this at a later time)

Currently the expiration is default at `DAY_IN_SECONDS`. This means all the queries for a site will be added to the map for 24 hours. Even if we solve the first problem and disable mapping for sites that have not opted in to "Use Object Cache", this is still a problem for sites that have opted-in (they're using POST requests, for example). 

For sites like clutchpoints that publish content constantly (200+ new articles each day), there's a high likelihood that the maps will grow significantly throughout a 24 hour period. 

If we reduce the expiration to something like 1 Hour, this will likely lead to smaller maps being produced and persisted.

We should make the default expiration time configurable via a Setting (similar to Max Age for network cache) but have a lower default.

### PR DESCRIPTION
**Pull request by jasonbahl**
- update testing workflow to test against WPGraphQL 1.14.4 (stop-gap as 1.14.5 had some changes that caused tests to break)
- added `Settings::cache_maps_enabled()` to be able to check if cache maps are enabled. Default is the same as "caching_enabled()" but can be filtered to be different (i.e. plugins that don't want to use "local" cache, but do want to track the mapping)
- Prevent maps from being built/stored if cache_maps are not enabled
- add filter to tests to enable cache maps (tests currently rely on the maps)

closes #229

### CODE REVIEW

File: src/Admin/Settings.php [hunk 1]
```diff
@@ -23,8 +23,29 @@ public static function caching_enabled() {
 		// get the cache_toggle setting
 		$option = function_exists( 'get_graphql_setting' ) ? \get_graphql_setting( 'cache_toggle', false, 'graphql_cache_section' ) : false;
 
+		$enabled = ( 'on' === $option );
+
+		$enabled = apply_filters( 'wpgraphql_cache_wordpress_cache_enabled', (bool) $enabled );
```
Comment (markkelnar) [inline src/Admin/Settings.php:28]: Other filters in this plugin, start with 'graphql_cache_...'.

File: src/Admin/Settings.php [hunk 2]
```diff
@@ -23,8 +23,29 @@ public static function caching_enabled() {
...
+	 *
+	 * Default behavior is to only enable building and storage of the cache maps if "WordPress Cache" (non-network cache) is enabled, but this can be filtered to be enabled without WordPress cache being enabled.
+	 *
+	 * @return bool
+	 */
+	public static function cache_maps_enabled() {
+
+		// Whether "WordPress Cache" (object/transient) cache is enabled
+		$enabled = self::caching_enabled();
+		return (bool) apply_filters( 'wpgraphql_cache_enable_cache_maps', (bool) $enabled );
```
Comment (markkelnar) [inline src/Admin/Settings.php:48]: See name change suggestion for other filter above.

File: src/Admin/Settings.php (hunk 1 above)
Comment (jasonbahl) [reply src/Admin/Settings.php:28]: ah ya, and the actions are a mix of `wpgraphql_cache` and `graphql_cache`

I'll change this, and make an issue to follow up and make things consistent across the board. 

I noticed the action `wpgraphql_cache_save_request` and changed these filters to follow that pattern, blindly assuming we were already consistent 😆

Comment (jasonbahl) [reply src/Admin/Settings.php:28]: Opened this issue to follow-up: https://github.com/wp-graphql/wp-graphql-smart-cache/issues/231

Comment (jasonbahl) [reply src/Admin/Settings.php]: @markkelnar ah yes, this is why I changed it to `wpgraphql_cache`: https://github.com/wp-graphql/wp-graphql-smart-cache/actions/runs/5555942329/jobs/10147786949?pr=230#step:7:20

Comment (jasonbahl) [reply src/Admin/Settings.php]: I reverted to `wpgraphql_cache` to be inline with the linter rules we have in place. 

I also updated #231 to remind us to consider the linter when we follow up to make all filter/action naming conventions the same

File: tests/wpunit/CacheInvalidationTest.php [hunk 3]
```diff
@@ -9,8 +9,12 @@ class CacheInvalidationTest extends \Codeception\TestCase\WPTestCase {
 	protected $collection;
 
 	public function setUp(): void {
+
 		\WPGraphQL::clear_schema();
 
+		// enable graphql cache maps
+		add_filter( 'graphql_cache_enable_cache_maps', '__return_true' );
```
Comment (markkelnar) [inline tests/wpunit/CacheInvalidationTest.php:16]: Is different filter name intentional?  Doesn't have the wp?

**Review by markkelnar — APPROVED** [review]

**Pull request merged**